### PR TITLE
Updates and improvements to image uploading

### PR DESCRIPTION
**Pull request by olcplif**
The text is checked for uploaded images when an article is created or updated. For this, regex checking is used. The IDs of the images are extracted. The article's ID is written in the records of these images in the DB.

As a result, the attached images are also deleted when the article is deleted.

I used Service Objects in the controller to upload the image.

- [x] use Service Objects to upload the image
- [x] fill the URL field in the DB record with an image's URL
- [x] check the article's text for uploaded images and fill article_id in the image's record in the DB
- [x] add tests

[https://trello.com/c/1b5q0WU3/79-fix-ckeditor](https://trello.com/c/1b5q0WU3/79-fix-ckeditor)

### CODE REVIEW

File: app/services/ck_editor_image_service.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class CkEditorImageService
+  def initialize(params)
```
Comment (mmrshk) [inline app/services/ck_editor_image_service.rb:4]: rename params to file

File: app/services/ck_editor_image_service.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    @file = params
+  end
+
+  def save
+    @uploaded_image = CkEditorImage.new(file: @file)
+    update_image_url if @uploaded_image.save
+  end
+
+  def file
```
Comment (mmrshk) [inline app/services/ck_editor_image_service.rb:13]: do you really need this method?

File: app/services/ck_editor_image_service.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    @uploaded_image.file
+  end
+
+  def errors
```
Comment (mmrshk) [inline app/services/ck_editor_image_service.rb:17]: and this?

File: app/services/ck_editor_image_service.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class CkEditorImageService
+  def initialize(params)
+    @file = params
+  end
+
+  def save
+    @uploaded_image = CkEditorImage.new(file: @file)
```
Comment (mmrshk) [inline app/services/ck_editor_image_service.rb:9]: empty space after

File: app/services/ck_editor_image_service.rb (hunk 1 above)
Comment (olcplif) [reply app/services/ck_editor_image_service.rb]: fixed

File: app/services/ck_editor_image_service.rb (hunk 2 above)
Comment (olcplif) [reply app/services/ck_editor_image_service.rb]: fixed

File: app/services/ck_editor_image_service.rb (hunk 3 above)
Comment (olcplif) [reply app/services/ck_editor_image_service.rb]: fixed

File: app/services/ck_editor_image_service.rb (hunk 4 above)
Comment (olcplif) [reply app/services/ck_editor_image_service.rb:11]: fixed

**Review by mmrshk — CHANGES_REQUESTED** [review]
We already have images for tests here -` spec/files`

**Comment by olcplif** [comment]
> We already have images for tests here -` spec/files`

Yes, but I needed an invalid file to check that it would not be saved to DB

**Review by mmrshk — CHANGES_REQUESTED** [review]
Why this branch includes changes from other PR's???

**Pull request merged**